imagersite/imager_images/tests.py

```diff
@@ -0,0 +1,242 @@
+from django.contrib.auth.models import User
+from imager_images.models import Photo, Album
+import factory
+
+# Create your tests here.
+
+
+class UserTestCase(TestCase):
+    """The User Model test class."""
+
+    class UserFactory(factory.django.DjangoModelFactory):
+        """Create test instance of User Class."""
+
+        class Meta:
+            """Invoke user instance using User model class."""
+
+            model = User
+
+        username = factory.Sequence(lambda n: "Photographer {}".format(n))
+        email = factory.LazyAttribute(
+            lambda x: "{}@theshot.com".format(x.username.replace(" ", ""))
+        )
+
+    class PhotoFactory(factory.django.DjangoModelFactory):
+        """Create test instance of Photo Class."""
+
+        class Meta:
+            """Invoke photo instance using Photo model class."""
+
+            model = Photo
+
+        title = factory.Sequence(lambda n: "Photo{}".format(n))
+
+    class AlbumFactory(factory.django.DjangoModelFactory):
+        """Create test instance of Album."""
+
+        class Meta:
+            """Invoke Album instance using Album model class."""
+
+            model = Album
+
+        title = factory.Sequence(lambda n: "Album{}".format(n))
+
+    def setUp(self):
+        """The setup and buildout for uses, photos, and albums."""
+        self.users = [self.UserFactory.create() for i in range(20)]
+        self.photos = [self.PhotoFactory.create() for i in range(20)]
+        self.albums = [self.AlbumFactory.create() for i in range(20)]
+
+    def test_photo_exists(self):
+        """Test existance of a photo."""
+        this_photo = self.photos[0]
+        this_photo.save()
+        self.assertTrue(self.photos[0])
+
+    def test_photo_has_a_title(self):
+        """Test photo had a title (built in tes."""
+        this_photo = self.photos[0]
+        this_photo.save()
+        self.assertTrue(self.photos[0].title)
+
+    def test_photo_has_a_photographer(self):
+        """Test photo has a photographer."""
+        this_photo = self.photos[0]
+        this_photo.save()
+        self.assertFalse(self.photos[0].photographer)
+        this_user = self.users[0]
+        this_user.save()
+        this_photo.photographer = this_user.profile
+        self.assertTrue(self.photos[0].photographer)
+
+    def test_album_has_a_description(self):
+        """Test an album has a description."""
+        this_album = self.albums[0]
+        this_album.description = "My Album"
+        this_album.save()
+        self.assertTrue(self.albums[0].description == "My Album")
+
+    def test_album_has_an_owner(self):
+        """Test album has a photographer, which is tied to a user."""
+        this_user = self.users[0]
+        this_user.username = "User 1"
+        this_user.save()
+        this_album = self.albums[0]
+        this_album.owner = this_user.profile
+        this_album.save()
+        self.assertTrue(self.albums[0].owner.user.username == "User 1")
+
```

Comment:
Excellent job testing all of these attributes on the models.  I think to save yourself some time/space, you could make this a little easier by adding all of the attributes into a list of tuples, with an input value, and you could just write one test to test them all.  Of course, you wouldn't have that large number of tests that you used to have but...